### DISCUSSION TITLE
Windows AOT

### DISCUSSION THREAD

**Discussion by gdamino** [dotnet/maui#24443] [General]
We have an app published for Android and IPhone. 
Now we want to publish for Windows Store but we noticed that the native compilation is not supported for that platform, the result is a package where you can just decompile all the IL code in just 3 seconds and have it clear and complete ready to use.
Native code is much more difficult to re-engineering.

So, I want to understand if there is a way to publish to Windows Store a MAUI App compiled to native code, someone can point me to a solution?

Thanks in advice

**Comment by MartyIX** [comment]
I believe that Windows AOT will be available in Windows App SDK 1.6 (https://learn.microsoft.com/en-gb/windows/apps/windows-app-sdk/preview-channel#native-aot-support-1), MAUI currently uses 1.5.

There is an open PR for bumping WASDK to 1.6 in MAUI here https://github.com/dotnet/maui/pull/24266 (draft!) but it's only for .NET 9. Though, you can probably achieve the same thing for .NET 8 with some work.

So the answer is "not yet", I'm afraid.

**Comment by gdamino** [comment]
I will try to make it working, or at least start to try it in .NET 9 preview.
I think is an important feature to compile AOT also for Windows and it's strange that only very few people request it.

**Reply by MartyIX** [reply]
If you could share your experience, it would be great. I'm waiting for the feature as well.

**Reply by gdamino** [reply]
Yes, for sure, but I think the MAUI team need to take this seriously and make it production-ready done because performance between IL and native code are very different and also it's another deterrent against code re-engineering.

**Reply by jingliancui** [reply]
I am also looking forward to this aot feature for maui win platform as of now winui 1.6 released.
